src/ghidra/p_code_extractor/PcodeExtractor.java

```diff
@@ -1,288 +1,249 @@
+                    context);
+            cconv.setFloatParameterRegister(float_register);
 
-        for (Instruction instr : instructions) {
-            PcodeBlockData.instruction = instr;
-            analysePcodeBlockOfAssemblyInstruction();
-            PcodeBlockData.instructionIndex++;
-        }
+            VarnodeSimple integer_return_register = get_integer_return_register(prototypeModel, ghidraProgram,
+                    context);
+            cconv.setIntegerReturnRegister(integer_return_register);
+
+            VarnodeSimple float_return_register = get_float_return_register(prototypeModel, ghidraProgram, context);
+            cconv.setFloatReturnRegister(float_return_register);
 
-        if (PcodeBlockData.blocks.get(0).getTerm().getDefs().isEmpty() && PcodeBlockData.blocks.get(0).getTerm().getJmps().isEmpty()) {
-            handleEmptyBlock(codeBlock);
+            callingConventions.put(prototypeModel.getName(), cconv);
         }
 
-        return PcodeBlockData.blocks;
-    }
+        // assembling everything together
+        ProjectSimple project = new ProjectSimple(functions, registerProperties, cpuArch, stackPointerRegister,
+                callingConventions, dataTypeProperties, imageBase);
 
+        // serialization
 
-    /**
-     * 
-     * @param codeBlock: Current empty block
-     * @return New jmp term containing fall through address
-     * 
-     * Adds fallthrough address jump to empty block if available
-     */
-    protected void handleEmptyBlock(CodeBlock codeBlock) {
+        Gson gson = new GsonBuilder().setPrettyPrinting().serializeNulls().create();
         try {
-            CodeBlockReferenceIterator destinations = codeBlock.getDestinations(getMonitor());
-            if(destinations.hasNext()) {
-                Tid jmpTid = new Tid(String.format("instr_%s_%s", codeBlock.getFirstStartAddress().toString(), 0), codeBlock.getFirstStartAddress().toString());
-                Tid gotoTid = new Tid();
-                String destAddr = destinations.next().getDestinationBlock().getFirstStartAddress().toString();
-                gotoTid.setId(String.format("blk_%s", destAddr));
-                gotoTid.setAddress(destAddr);
-                PcodeBlockData.blocks.get(0).getTerm().addJmp(new Term<Jmp>(jmpTid, new Jmp(ExecutionType.JmpType.GOTO, "BRANCH", new Label((Tid) gotoTid), 0)));
-            }
-        } catch (CancelledException e) {
-            System.out.printf("Could not retrieve destinations for block at: %s\n", codeBlock.getFirstStartAddress().toString());
+            FileWriter writer = new FileWriter(getScriptArgs()[0]);
+            gson.toJson(project, writer);
+            writer.close();
+        } catch (JsonIOException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
         }
-    }
 
+        println("Pcode was successfully extracted!");
+    }
```

Comment:
If an error is catched here, then we should not print the "success" message. Also: Are there string error messages that we can print together with the stack traces here?

Reply:
Good catch!